src/utils/numericLiterals.ts

```diff
@@ -41,12 +46,13 @@ export const isWellFormedInteger = (s: string) =>
 const integerLikeMatcher = /^-?\d+$/
 const isIntegerLike = (s: string) => integerLikeMatcher.test(s)
 
-type NumericLiteralKind = "number" | "bigint" | "integer"
+type NumericLiteralKind = "number" | "bigint" | "integer" | "Date"
```

Comment:
I don't think a Date would be considered a numeric literal, so this file shouldn't actually be changed at all. We should just be able to rely on the simple "isValidDate" function you wrote somewhere else.